### PR TITLE
feat: remark-mdx is not compatible with normal markdown syntaxes

### PR DESCRIPTION
**Pull request by JounQin**
As `remark-mdx@v2` is [not compatible](https://github.com/mdx-js/mdx/issues/1042) with normal markdown comment syntaxes anymore, so markdown and mdx files should be processed specifically.

@johno @wooorm

### CODE REVIEW

**Comment by JounQin** [comment]
This branch is not required to merge for now, but would be better to publish a `next` tag for [mdx-js/mdx](https://github.com/mdx-js/mdx) itself.

**Comment by wooorm** [comment]
Awesome work @JounQin! The PR is currently marked as draft. Does that mean that you‘re still working on it, and/or that it’s waiting for MDX@2?

**Comment by JounQin** [comment]
The fix should be compatible between v1 and v2, so I think it's ready to merge!

File: packages/eslint-mdx/src/parser.ts [hunk 1]
```diff
@@ -322,12 +323,20 @@ export class Parser {
 
     const value = node.value as string
 
+    const { loc, start, end } = normalizePosition(node.position)
+
     // fix #4
     if (isComment(value)) {
+      const comment = COMMENT_CONTENT_REGEX.exec(value)[2]
+      this._ast.comments.push({
```
Comment (wooorm) [inline packages/eslint-mdx/src/parser.ts:331]: Does `_ast` here point to the whole ES AST? If so, it won’t work if a part but not the whole document is ignored maybe?

https://github.com/remarkjs/remark-message-control#markers

Comment (JounQin) [reply packages/eslint-mdx/src/parser.ts:331]: It is only for `<!-- eslint-disable ruleId -->`.

**Pull request merged**